File: modelInOtherObjectOrientedLanguages/ruby/MARK_II/Cell.rb
class Cell
  protected @isActive
  protected @wasActive

  # self makes it a class Cell method
  def public self.getActiveState?
  	return @isActive
  end

  def public self.setActiveState=(value)
  	@isActive = value
  end

  def public self.getPreviousActiveState?
  	return @wasActive
  end

  def public self.setPreviousActiveState=(value)
  	@wasActive = value
  end
end

File: modelInOtherObjectOrientedLanguages/javascript/MARK_II/Cell.js
function Cell(isActive, wasActive) {
    this.isActive = isActive || null;
    this.wasActive = wasActive || null;
}

RiddleInfo.prototype.getActiveState = function() {
    return this.isActive;
}

RiddleInfo.prototype.setActiveState = function(activeState) {
    this.isActive = activeState;
}

RiddleInfo.prototype.getPreviousActiveState = function() {
    return this.wasActive;
}

RiddleInfo.prototype.setPreviousActiveState = function(previousActiveState) {
    this.wasActive = previousActiveState;
}
  
RiddleInfo.prototype.fill = function(newCellProperties) {
    for (var cellProperty in newCellProperties) {
        if (this.hasOwnProperty(cellProperty) && newCellProperties.hasOwnProperty(cellProperty)) {
            if (this[cellProperty] !== 'undefined') {
                this[cellProperty] = newCellProperties[cellProperty];
            }
        }
    }
}; 

module.exports = Cell;